File: MusicPlayerApp/app/src/main/java/com/example/musicplayerapp/MainActivity2.java
package com.example.musicplayerapp;
import android.os.Bundle;
import android.content.Intent;
import androidx.appcompat.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;
public class MainActivity2 extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Button startMusicButton = findViewById(R.id.startMusicButton);
        Button stopMusicButton = findViewById(R.id.stopMusicButton);
        startMusicButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity2.this,
                        MusicService2.class);
                startService(intent); // Start the foreground service
            }
        });
        stopMusicButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity2.this,
                        MusicService2.class);
                stopService(intent); // Stop the foreground service
            }
        });
    }
}

File: CustomBroadcastReceiverApp/app/src/main/java/com/example/custombroadcastreceiverapp/MainActivity2.kt
package com.example.custombroadcastreceiverapp

import android.content.Intent
import android.content.IntentFilter
import android.os.Bundle
import android.widget.Button
import androidx.activity.enableEdgeToEdge
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.appcompat.app.AppCompatActivity
import androidx.localbroadcastmanager.content.LocalBroadcastManager


class MainActivity2 : AppCompatActivity(){

    private lateinit var myReceiver: MyCustomBroadcastReceiver2

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)

        //myReceiver = MyCustomBroadcastReceiver2()

        //myReceiver = MyBroadcast()
        val sendBroadcastButton: Button = findViewById(R.id.sendBroadcastButton)

        sendBroadcastButton.setOnClickListener {
            // Create an intent with the custom action
            val intent = Intent("com.example.CUSTOM_ACTION")
            // Send the broadcast
            LocalBroadcastManager.getInstance(this).sendBroadcast(intent)
        }

        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

    }
    override fun onStart() {
        super.onStart()
        // Register the receiver with the intent filter
        LocalBroadcastManager.getInstance(this)
            .registerReceiver(myReceiver,
                IntentFilter("com.example.CUSTOM_ACTION"))
    }
    override fun onStop() {
        super.onStop()

        // Unregister the receiver to prevent memory leaks

        LocalBroadcastManager.getInstance(this).unregisterReceiver(myReceiver)
    }


}
